recipes/qt/all/conanfile.py

```diff
@@ -0,0 +1,685 @@
+                    "armv7hf": "qnx-armle-v7-qcc",
+                    "armv7s": "qnx-armle-v7-qcc",
+                    "armv7k": "qnx-armle-v7-qcc",
+                    "x86": "qnx-x86-qcc",
+                    "x86_64": "qnx-x86-64-qcc"}.get(str(self.settings.arch))
+        elif self.settings.os == "Emscripten" and self.settings.arch == "wasm":
+            return "wasm-emscripten"
+
+        return None
+
+    def build(self):
+        if self.settings.compiler in ["gcc", "clang"]:
+            if tools.Version(self.settings.compiler.version) < "5.0":
+                raise ConanInvalidConfiguration("qt 5.15.0 is not support on GCC or clang before 5.0")
+        args = ["-confirm-license", "-silent", "-nomake examples", "-nomake tests",
+                "-prefix %s" % self.package_folder]
+        args.append("-v")
+        args.append("-archdatadir  %s" % os.path.join(self.package_folder, "bin", "archdatadir"))
+        args.append("-datadir  %s" % os.path.join(self.package_folder, "bin", "datadir"))
+        args.append("-sysconfdir  %s" % os.path.join(self.package_folder, "bin", "sysconfdir"))
+        if self.options.commercial:
+            args.append("-commercial")
+        else:
+            args.append("-opensource")
+        if not self.options.GUI:
+            args.append("-no-gui")
+        if not self.options.widgets:
+            args.append("-no-widgets")
+        if not self.options.shared:
+            args.insert(0, "-static")
+            if self.settings.compiler == "Visual Studio":
+                if self.settings.compiler.runtime == "MT" or self.settings.compiler.runtime == "MTd":
+                    args.append("-static-runtime")
+        else:
+            args.insert(0, "-shared")
+        if self.options.multiconfiguration:
+            args.append("-debug-and-release")
+        elif self.settings.build_type == "Debug":
+            args.append("-debug")
+        elif self.settings.build_type == "Release":
+            args.append("-release")
+        elif self.settings.build_type == "RelWithDebInfo":
+            args.append("-release")
+            args.append("-force-debug-info")
+        elif self.settings.build_type == "MinSizeRel":
+            args.append("-release")
+            args.append("-optimize-size")
+
+        for module in self._submodules:
+            if module != 'qtbase' and not getattr(self.options, module) \
+                    and os.path.isdir(os.path.join(self.source_folder, 'qt5', self._submodules[module]['path'])):
+                args.append("-skip " + module)
+
+        args.append("--zlib=system")
+
+        # openGL
+        if self.options.opengl == "no":
+            args += ["-no-opengl"]
+        elif self.options.opengl == "es2":
+            args += ["-opengl es2"]
+        elif self.options.opengl == "desktop":
+            args += ["-opengl desktop"]
+        elif self.options.opengl == "dynamic":
+            args += ["-opengl dynamic"]
+        
```

Comment:
Nit: This line contains trailing whitespaces in the file. Not visible on github for unknown reasons.